RawBencher/Benchers/LINQ2DBNormalBencher.cs

```diff
@@ -1,47 +1,153 @@
+        {
+            using (var db = new Db(ConnectionString))
+            {
+                return (from soh in db.SalesOrderHeaders
+                               .LoadWith(x => x.SalesOrderDetails)
+                               .LoadWith(x => x.Customer)
+                        where soh.SalesOrderID > 50000 && soh.SalesOrderID <= 51000
+                        select soh)
+                             .ToList();
+            }
+
+        }
+        public override async Task<IEnumerable<LINQ2DB.Bencher.SalesOrderHeader>> FetchGraphAsync()
+        {
+            using (var db = new Db(ConnectionString))
+            {
+                return await (from soh in db.SalesOrderHeaders
+                                .LoadWith(x => x.SalesOrderDetails)
+                                .LoadWith(x => x.Customer)
+                              where soh.SalesOrderID > 50000 && soh.SalesOrderID <= 51000
+                              select soh)
+                    .ToListAsync();
+            }
+        }
+
+        public override void VerifyGraphElementChildren(LINQ2DB.Bencher.SalesOrderHeader parent, BenchResult resultContainer)
+        {
+            int amount = 0;
+            foreach (var sod in parent.SalesOrderDetails)
+            {
+                if (sod.SalesOrderID > 0)
+                {
+                    amount++;
+                }
+                else
+                {
+                    return;
+                }
+            }
+
+            resultContainer.IncNumberOfRowsForType(typeof(SalesOrderDetail), amount);
+
+            if ((parent.Customer == null) || (parent.Customer.CustomerID <= 0))
+            {
+                return;
+            }
+
+            resultContainer.IncNumberOfRowsForType(typeof(Customer), 1);
+        }
+
+        public override IEnumerable<CreditCard> CreateSetForInserts(int amountToInsert)
+        {
+            var toReturn = new CreditCard[amountToInsert];
+
+            for (int i = 0; i < amountToInsert; i++)
+            {
+                toReturn[i] = new CreditCard()
+                {
+                    CardNumber = Guid.NewGuid().ToString("N").Substring(0, 24),
+                    CardType = "Vista",
+                    ExpMonth = 11,
+                    ExpYear = 2018,
+                    ModifiedDate = DateTime.Now
+                };
+            }
+
+            return toReturn;
+        }
+
+        protected override IEnumerable<CreditCard> FetchInserted(int amountInserted)
+        {
+            using (var db = new Db(ConnectionString))
+            {
+                return db.CreditCards
+                    .Where(_ => _.CreditCardID > 19237)
+                   .ToList();
+            }
+        }
+
+        protected override void DeleteInserted(IEnumerable<CreditCard> toDelete)
+        {
+            using (var db = new Db(ConnectionString))
+            {
+                // other providers cheat by using "CreditCardId > 19237" filter
```

Comment:
It's not used in measurements, so you can use whatever you want to get the rows as quickly as possible from the DB. So if a single delete with a single predicate does the trick, that's fine, the faster the better as e.g. with EF it really slows things down as they have to fetch each entity that's to be deleted.